Assets/Scripts/Interactables/Dialogue/DialogueInteractable.cs

```diff
@@ -11,29 +11,77 @@ public class DialogueInteractable : Interactable
 
     protected bool isConversing = false;
 
-    private int activeLineIndex = 0;
+    private int activeLineIndex = 0;
+
+
+    private bool autoPlaying = false;
 
-    public override void OnInteract()  
+    public float waitLineTime = 2.0f;
+
+    protected virtual void Start()
+    {
+        speakerUINPC = Utils.GetRequiredComponentInChildren<SpeakerUI>(this);
+    }
+
+    // Conversation to happen when interacted with
+    public override void OnInteract()
     {
-        if (!isConversing)
+        if (!autoPlaying)
         {
-            isConversing = true;
+            if (!isConversing)
+            {
+                isConversing = true;
 
-            // Freeze player when conversing
-            player.GetComponent<PlayerController>().CanMove = false;
+                speakerUIBirdie = Utils.GetRequiredComponentInChildren<SpeakerUI>(player);
 
-            speakerUIBirdie = Utils.GetRequiredComponentInChildren<SpeakerUI>(player);
-            speakerUINPC = Utils.GetRequiredComponentInChildren<SpeakerUI>(this);
+                // Freeze player when conversing
+                player.GetComponent<PlayerController>().CanMove = false;
 
-            HideInteractUI();
-            AdvanceConversation();
+                HideInteractUI();
+                AdvanceConversation();
 
-            base.OnInteract();
+                base.OnInteract();
+            }
+            else
+            {
+                AdvanceConversation();
+            }
         }
         else
         {
-            AdvanceConversation();
+            StartCoroutine(AutoplayConversations()); ;
         }
+    }
+
+    // Functionality for Autoplay Conversation to be triggered
+    public virtual void TriggerAutoplay(GameObject target)
+    {
+        // Possible issue here when target is not Birdie
+        speakerUIBirdie = Utils.GetRequiredComponentInChildren<SpeakerUI>(target);
+        autoPlaying = true;
+    }
+
+    protected virtual void OnAutoplayComplete()
+    {
+        autoPlaying = false;
+    }
+
+
+    IEnumerator AutoplayConversations()
+    {
+        while (activeLineIndex < conversation.lines.Length)

```

Comment:
Just my own personal preference but I think I'd prefer doing `activeLineIndex = 0;` before the while loop, rather than after (who knows if someone else will change the value from 0 before calling this function?).